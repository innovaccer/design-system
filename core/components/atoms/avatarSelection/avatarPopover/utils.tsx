import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
