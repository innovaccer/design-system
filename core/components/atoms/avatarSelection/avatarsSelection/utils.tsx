import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
