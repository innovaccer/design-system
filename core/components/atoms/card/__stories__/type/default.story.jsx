import { Card } from '@/index';

// CSF format story
export const defaultCard = () => {
  return (
    <Card>
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Default Card',
  component: Card,
};
