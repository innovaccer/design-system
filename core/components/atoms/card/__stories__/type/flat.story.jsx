import { Card } from '@/index';

// CSF format story
export const Flat = () => {
  return (
    <Card shadow="none">
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Flat',
  component: Card,
};
