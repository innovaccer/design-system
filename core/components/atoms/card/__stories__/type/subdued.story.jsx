import { Card, CardSubdued } from '@/index';

// CSF format story
export const Subdued = () => {
  return (
    <CardSubdued>
      <div className="p-8" />
    </CardSubdued>
  );
};

export default {
  title: 'Components/Card/Type/Subdued',
  component: Card,
  subcomponents: { CardSubdued },
  parameters: {
    docs: {
      docPage: {
        title: 'CardSubdued',
        description: 'A component to highlight section of a card.',
      },
    },
  },
};
