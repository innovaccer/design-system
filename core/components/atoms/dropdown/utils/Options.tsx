export let dropdownOptions: any[] = [];

for (let i = 1; i <= 50; i++) {
  dropdownOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
  });
}

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = dropdownOptions.slice(0, 10);

export const storySections = { 0: 'subheading', 5: 'subheading' };

export const subInfoItems = [
  {
    label: 'Option1',
    value: 'Option1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option2',
    value: 'Option2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option3',
    value: 'Option3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option1',
    value: 'Option1',
    icon: 'events'
  },
  {
    label: 'Option2',
    value: 'Option2',
    icon: 'events',
  },
  {
    label: 'Option3',
    value: 'Option3',
    icon: 'events',
  },
];
