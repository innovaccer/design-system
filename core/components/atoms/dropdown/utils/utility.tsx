const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: any) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const getOptions = (offset: number, limit: number, searchTerm: string, options: any) => {
  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
  return new Promise(resolve => {
    resolve({
      offset,
      slicedOptions: searchedOptions.slice(offset, offset + limit),
      options: searchedOptions,
    });
  });
};

export const getValuesFromSelectedObj = (selectedArray: any[] = []) => {
  const selectedValues: any[] = [];

  selectedArray.forEach((selectedObj: any) => {
    const { value } = selectedObj;
    selectedValues.push(value);
  });

  return selectedValues;
};

export const getIndexesFromSelectedObj = (selectedArray: any[] = []) => {
  return (Array.from(Array(selectedArray.length).keys()));
};

export const getValuesFromInd = (selectedArray: number[], options: any[]) => {
  const result: any[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].value);
  });

  return result;
};

export const getLabelsFromInd = (selectedArray: number[], options: any[]) => {
  const result: string[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].label);
  });

  return result;
};
