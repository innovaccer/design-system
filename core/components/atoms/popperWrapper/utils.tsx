// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
