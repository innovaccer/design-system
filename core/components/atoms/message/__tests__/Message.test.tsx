import * as React from 'react';
import { render } from '@testing-library/react';
import { Message, Text } from '@/index';
import { MessageProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const appearances: Props['appearance'][] = ['alert', 'info', 'warning', 'success'];
const title = 'Title goes here';
const description = 'Description goes here';
const actions = (
  <>
    <Text className="mr-5 cursor-pointer" appearance="link">
      Action 1
    </Text>
    <Text className="cursor-pointer" appearance="link">
      Action 2
    </Text>
  </>
);

describe('Message component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearances, { required: true, iterate: true }),
    title: valueHelper(title, { required: true }),
    description: valueHelper(description, { required: true }),
    actions: valueHelper(actions, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Message {...attr}>Description goes here</Message>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Message component', () => {
  it('renders description', () => {
    const { getByTestId } = render(<Message appearance="info" description={description} />);
    expect(getByTestId('DesignSystem-Message--Description').textContent).toMatch(description);
  });
});

describe('Message Component with overwrite class', () => {
  it('overwrite Message class', () => {
    const { getByTestId } = render(<Message className="MessageClass" description={description} />);
    expect(getByTestId('DesignSystem-Message')).toHaveClass('MessageClass');
  });
});

describe('Message component with prop:Title', () => {
  it('renders title', () => {
    const { getByTestId } = render(<Message appearance="info" title={'Masala'} description={description} />);
    expect(getByTestId('DesignSystem-Message--Title').textContent).toMatch('Masala');
    expect(getByTestId('DesignSystem-Message--Icon')).toHaveClass('Message-icon--withTitle');
  });

  it('should not have Message-title class if title is not present ', () => {
    const { getByTestId, queryByTestId } = render(<Message appearance="info" description={description} />);
    expect(queryByTestId('DesignSystem-Message--Title')).not.toBeInTheDocument();
    expect(getByTestId('DesignSystem-Message--Icon')).not.toHaveClass('Message-icon--withTitle');
  });
});

describe('Message component with prop:appearance', () => {
  appearances.forEach((appearance) => {
    it(`should have the Message--${appearance} class when appearance = ${appearance}`, () => {
      const { getByTestId } = render(<Message appearance={appearance} description={description} />);
      expect(getByTestId('DesignSystem-Message')).toHaveClass(`Message--${appearance}`);
    });
  });
});

describe('Message component with prop: actions', () => {
  it('renders actions', () => {
    const { getByTestId } = render(<Message appearance="info" description={description} actions={actions} />);
    expect(getByTestId('DesignSystem-Message--actions')).toBeInTheDocument();
  });
});
