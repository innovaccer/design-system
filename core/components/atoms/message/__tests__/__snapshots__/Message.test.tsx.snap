// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component 
  appearance: "alert", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--alert"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-icons material-icons-round Icon Icon--alert Message-icon Message-icon--alert Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        style="font-size: 16px; width: 16px;"
      >
        error_round
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading Message-heading--alert"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text Message-text--alert"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "info", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--info"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-icons material-icons-round Icon Icon--info Message-icon Message-icon--info Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        style="font-size: 16px; width: 16px;"
      >
        info_round
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading Message-heading--info"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text Message-text--info"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "success", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--success"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-icons material-icons-round Icon Icon--success Message-icon Message-icon--success Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        style="font-size: 16px; width: 16px;"
      >
        check_circle_round
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading Message-heading--success"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text Message-text--success"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "warning", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--warning"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-icons material-icons-round Icon Icon--warning Message-icon Message-icon--warning Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        style="font-size: 16px; width: 16px;"
      >
        warning_round
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading Message-heading--warning"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text Message-text--warning"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
